syntax = "proto3";

option go_package = "pb/game";

import "proto/common/tip.proto";
import "proto/common/user_accounts.proto";

message AccountSimplePlayerWrapper
{
  AccountSimplePlayer player = 1;
}

message LoginRequest
{
  string account = 1;
  string password = 2;
}

message LoginResponse
{
  TipInfoMessage error_message = 1;
  repeated AccountSimplePlayerWrapper players = 2;
}

message TestResponse
{
  TipInfoMessage error_message = 1;
  repeated AccountSimplePlayerWrapper players = 2;
  repeated string teststring = 3;
  repeated int32 testint = 4;
}

message CreatePlayerRequest
{
	
}

message CreatePlayerResponse
{
	TipInfoMessage error_message = 1;
	repeated AccountSimplePlayerWrapper players = 2;
}

message EnterGameRequest
{
	uint64 player_id  = 1;
}

message EnterGameResponse
{
	TipInfoMessage error_message = 1;
}

message LeaveGameRequest
{
}

enum ClientMessageType
{
  CLIENT_REQUEST = 0;
  CLIENT_RESPONSE = 1;
  CLIENT_RPC_ERROR = 2; // not used
}

enum TipCode
{
  CLIENT_RPC_NO_ERROR = 0;
  CLIENT_WRONG = 1;
  CLIENT_NO_SERVICE = 2;
  CLIENT_NO_METHOD = 3;
  CLIENT_INVALID_REQUEST = 4;
  CLIENT_INVALID_RESPONSE = 5;
  CLIENT_TIMEOUT = 6;
}

message ClientRequest
{
  uint64 id = 1;
  string service = 2;
  string method = 3;
  bytes body = 4;
  uint32 message_id = 5;
}

